type_classes.md

```diff
@@ -206,20 +339,49 @@ With this added to the earlier instance declarations, type class instance can in
 # instance : Inhabited Nat where
 #  default := 0
 # constant arbitrary [Inhabited a] : a :=
-#  Inhabited.default
-instance [Inhabited a] [Inhabited b] : Inhabited (a × b) where
-  default := (arbitrary, arbitrary)
+#   Inhabited.default
+# instance [Inhabited a] [Inhabited b] : Inhabited (a × b) where
+#   default := (arbitrary, arbitrary)
 
 #eval (arbitrary : Nat × Bool)
 -- (0, true)
 # end Ex
 ```
-Similarly, we can inhabit type function with suitable constant functions:
+
+Note: Remember to use `\times` here and not the letter `x`. This will not work `Nat x Bool`.
+
+Similarly, we can inhabit the function type with suitable defaults:
 ```lean
+# export Inhabited (default)
 instance [Inhabited b] : Inhabited (a -> b) where
   default := fun _ => arbitrary
+
+#eval (default : Nat -> Bool) 5
+
+```
+
+BUGBUG: Why don't we also need to say a is inhabited?  Conversely, how is `a` even "defined"
```

Comment:
Ok, so why can't I use underscore instead of `a` which is the convention for "anything can go here":

```lean
instance [Inhabited b] : Inhabited (_ -> b) where
  default := fun _ => arbitrary
```

Reply:
And it is a single letter magic, this produces an error:

```lean
instance [Inhabited b] : Inhabited (elephant -> b) where      -- unknown identifier 'elephant'
  default := fun _ => arbitrary
```

Reply:
Yes this is something I've seen Lean do in the past as well...you can basically omit single letter names and it will add an implicit definition for them, in this case Lean will translate the instance to something like:
```lean
instance {a : Sort u} {b : Sort v} [Inhabited b] : Inhabited (a -> b) where
  default := fun _ => arbitrary
```
But for multi character names it will not do this automatically, I'm guessing this is to avoid errors where you are referring to identifiers that actually do exist but typoed or they don't exist in the current scope etc.